src/darwin/Framework/CHIPTests/MTRSwiftPairingTests.swift

```diff
@@ -0,0 +1,88 @@
+import Matter
+import XCTest
+
+// This more or less parallels the "no delegate" case in MTRPairingTests
+
+struct Constants {
+    static let localPort = 5541
+    static let vendorID = 0xFFF1
+    static let onboardingPayload = "MT:Y.K90SO527JA0648G00"
+    static let deviceID = 0x12344321
+    static let timeoutInSeconds : UInt16 = 3
+}
+
+class MTRSwiftPairingTestControllerDelegate : NSObject, MTRDeviceControllerDelegate {
+    let expectation: XCTestExpectation
+
+    init(withExpectation providedExpectation: XCTestExpectation) {
+        expectation = providedExpectation
+    }
+
+    @available(macOS, introduced: 13.3)
+    func controller(_ controller: MTRDeviceController, statusUpdate status: MTRCommissioningStatus) {
+        XCTAssertNotEqual(status, MTRCommissioningStatus.failed)
+    }
+
+    @available(macOS, introduced: 13.3)
+    func controller(_ controller: MTRDeviceController, commissioningSessionEstablishmentDone error: Error?) {
+        XCTAssertNil(error)
+
+        do {
+            try controller.commissionNode(withID: Constants.deviceID as NSNumber, commissioningParams: MTRCommissioningParameters())
+        } catch {
+            XCTFail("Could not start commissioning of node")
+        }
+
+        // Keep waiting for commissioningComplete
+    }
+
+    func controller(_ controller: MTRDeviceController, commissioningComplete error: Error?, nodeID: NSNumber?) {
+        XCTAssertNil(error)
+        XCTAssertEqual(nodeID, Constants.deviceID as NSNumber)
+        expectation.fulfill()
+    }
+}
+
+class MTRSwiftPairingTests : XCTestCase {
+    @available(macOS, introduced: 13.3)
+    func test001_BasicPairing() throws {
+        let factory = MTRDeviceControllerFactory.sharedInstance()
```

Comment:
so the (older) way I would do this: 
guard let blah = foo() else {
  XCTFail("msg")
  return
}

It's a bit clunky because you need the return in there as well as the fail.  The newer way to do this would be
let factory = try XCTUnwrap(foo())
which should throw an error if foo() returns nil.

Reply:
Ah, that's nice.  Thank you.

Reply:
Actually, given that sharedInstance is defined as not returning nil, I think we can just drop this nil-checking bit entirely.